### PR TITLE
fix $JAVA_HOME conflict in /etc/profile.d, add openjdk-profile trigger

### PR DESCRIPTION
**Pull request by classabbyamp**
#### Testing the changes
- I tested the changes in this PR: **YES**

[ci skip]

### CODE REVIEW

File: srcpkgs/xbps-triggers/files/openjdk-profile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+targets)
+	echo "post-install"
+	;;
+run)
+	[ -f "$jdkprofile" ] && exit 0
+
+	cat > "$jdkprofile" <<-EOF
+_java_bin_dir=\$(dirname \$(readlink -f /usr/bin/java))
+[ -x /usr/bin/java ] && export JAVA_HOME=\${JAVA_HOME:=\${_java_bin_dir%/bin}}
+unset _java_bin_dir
```
[the remark this answers is not in the document]
Comment (classabbyamp) [reply srcpkgs/xbps-triggers/files/openjdk-profile]: done

**Comment by q66** [comment]
unless you force every openjdk package to depend on at least some version of xbps-triggers this will break horribly

also still not sure if i'm a fan of this approach, we should be killing xbps-triggers, not extending it with more junk, the whole package is a poorly thought out hack that is 90% useless boilerplate

File: srcpkgs/xbps-triggers/files/openjdk-profile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	;;
+run)
+	[ -f "$jdkprofile" ] && exit 0
+
+	cat > "$jdkprofile" <<-'EOF'
+if _java_bin=$(readlink -e /usr/bin/java); then
+	_java_bin="${_java_bin%/bin/java}"
+	[ -d "${_java_bin}" ] && export JAVA_HOME="${JAVA_HOME:=${_java_bin}}"
+fi
+unset _java_bin
```
Comment (ahesford) [inline srcpkgs/xbps-triggers/files/openjdk-profile:31]: The `-` in `<<-'EOF'` throws away leading tabs so the profile snippet it writes is a bit ugly. I recommend a single tab-indent for the *entire* block, including closing `EOF`, and then adding an extra 2 or 4 spaces on the lines that should be indented in the output.

Alternatively, just drop the `-` and the formatting will be preserved.

**Review by ahesford — APPROVED** [review]
@q66 I agree that `xbps-triggers` is hacky, but it solves a real problem here using tooling that isn't going away.

The alternative is putting the generic profile script in an `openjdk-profile` package and having every JDK package depend on that, which would require people who want to override the profile to `ignorepkg` the profile and write their own. That seems less convenient than using the trigger.

File: srcpkgs/xbps-triggers/files/openjdk-profile (hunk 2 above)
Comment (classabbyamp) [reply srcpkgs/xbps-triggers/files/openjdk-profile:31]: done

**Comment by q66** [comment]
the xbps-triggers added dependency does not solve anything

it must be specifically `xbps-triggers>=THE_NEW_VERSION_THAT_ADDED_STUFF`

**Comment by classabbyamp** [comment]
xlint there should be ignored, the new version lint meant I needed to update the version line, but there was no update

**Review by ahesford — APPROVED** [review]
I am fine with this, but it was @q66 who objected to the use of xbps-triggers here.

**Comment by classabbyamp** [comment]
would an `openjdk-profile` package be an alternative solution?

**Comment by classabbyamp** [comment]
So should I switch this to the `openjdk-profile` package solution or something?

**Pull request merged**